### PR TITLE
biopython protein auxiliary tasks

### PR DESCRIPTION
**Pull request by karinazad**
## Description
Add transforms for biopython auxiliary tasks

## Type of Change
- [ ] Bug fix
- [x] New feature  
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Code refactoring

### CODE REVIEW

File: src/lobster/transforms/functional/_biopython.py [hunk 1]
```diff
@@ -1,30 +1,255 @@
...
+        instability_index, isoelectric_point, alpha_helix_fraction, turn_structure_fraction,
+        beta_sheet_fraction, molar_extinction_coefficient_reduced_cysteines,
+        molar_extinction_coefficient_oxidized_cysteines, grand_average_hydropathy_index,
+        net_charge_at_ph_6, net_charge_at_ph_7.
+    return_as_tensor : bool
+        If True, returns features as a tensor. If False, returns as a dictionary.
+        Defaults to False.
+    complex_separator : str | None
+        Separator used to split protein complexes into individual chains. If None,
+        treats the entire sequence as a single chain. Defaults to ".".
```
Comment (taylormjs) [inline src/lobster/transforms/functional/_biopython.py:161]: Curious, why is "." the default chain separator?

File: tests/lobster/transforms/test_biopython_transform.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        transform = ProteinToBioPythonFeaturesTransform(standardize=True)
+        out = transform("MALWMRLLPLLALLALWGPDPAAA")
+        assert out is not None
+        assert isinstance(out, Tensor)
+
+        assert torch.allclose(
+            out,
+            torch.tensor(
+                [
+                    0.9070,
```
Comment (taylormjs) [inline tests/lobster/transforms/test_biopython_transform.py:104]: Just a note, I wonder how much these would change if a different random seed was used for getting the mean and scale

File: src/lobster/callbacks/_auxiliary_task_loss_weight_scheduler.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        if steps_since_start >= self.ramp_steps:
+            return self.max_weight
+
+        # Progress through the ramp period
+        progress = steps_since_start / self.ramp_steps
+
+        if self.schedule_type == "linear":
+            weight = progress * self.max_weight
+        elif self.schedule_type == "cosine":
+            import math
```
Comment (taylormjs) [inline src/lobster/callbacks/_auxiliary_task_loss_weight_scheduler.py:65]: Move to top level import?

File: src/lobster/constants/_biopython_features.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+}
+
+PEPTIDE_WARNING_THRESHOLD = 50
+
+# Obtained by fitting a StandardScaler to the computed BioPython features of the PeptideAtlas dataset
+# Subset N=1,000,000, split=train, seed=0
+BIOPYTHON_PEPTIDE_SCALER_PARAMS = {
+    "alpha_helix_fraction": {
+        "mean": 0.3190,
+        "scale": 0.1436,
```
[the remark this answers is not in the document]
Comment (karinazad) [reply src/lobster/constants/_biopython_features.py:43]: yep! scale is used just because that's how they call it in https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.StandardScaler.html
